server.py

```diff
@@ -116,21 +119,23 @@ def validate():
         if schema is None:
             return client_error("Unsupported schema version '%s'" % version)
 
-        tx_id = None
         if 'tx_id' in json_data:
             tx_id = json_data['tx_id']
 
-        bound_logger = logger.bind(tx_id=tx_id)
+            bound_logger = logger.bind(tx_id=tx_id)
+        else:
+            bound_logger = logger.bind(survey_id=json_data.get('survey_id'))
+
```

Comment:
Would this do the whole job?

`bound_logger = logger.bind(survey_id=json_data.get('survey_id'), tx_id=json_data.get('tx_id'))`